File: server2/src/main/java/com/example/FirebaseConfig.java
package com.example;

import com.google.auth.oauth2.GoogleCredentials;
import com.google.firebase.FirebaseApp;
import com.google.firebase.FirebaseOptions;
import com.google.firebase.auth.FirebaseAuth;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import java.io.FileInputStream;
import java.io.IOException;

@Configuration
class FirebaseConfig {
    @Bean
    public FirebaseAuth firebaseAuth() throws IOException {
        FileInputStream serviceAccount = new FileInputStream("firebase-adminsdk.json");

        FirebaseOptions options = new FirebaseOptions.Builder()
                .setCredentials(GoogleCredentials.fromStream(serviceAccount))
                .setDatabaseUrl("https://XXX.firebaseio.com/").build();

        FirebaseApp.initializeApp(options);

        return FirebaseAuth.getInstance();
    }
}

File: server/src/main/kotlin/com/va034600/security/FirebaseConfig.kt
package com.va034600.security

import com.google.auth.oauth2.GoogleCredentials
import com.google.firebase.FirebaseApp
import com.google.firebase.FirebaseOptions
import com.google.firebase.auth.FirebaseAuth
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import java.io.FileInputStream
import java.io.IOException

@Configuration
class FirebaseConfig {
    @Bean
    @Throws(IOException::class)
    fun firebaseAuth(): FirebaseAuth {
        val serviceAccount = FileInputStream("firebase-adminsdk.json")
        val options = FirebaseOptions.Builder()
                .setCredentials(GoogleCredentials.fromStream(serviceAccount))
                .setDatabaseUrl("https://XXX.firebaseio.com/").build()

        FirebaseApp.initializeApp(options)

        return FirebaseAuth.getInstance()
    }
}
